src/solvers/conflict_provider.h

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************\
+
+Module: Capability to check whether an expression is a reason for
+        the solver returning UNSAT
+
+Author: Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// Capability to check whether an expression is a reason for
+/// the solver returning UNSAT
+
+#ifndef CPROVER_SOLVERS_CONFLICT_PROVIDER_H
+#define CPROVER_SOLVERS_CONFLICT_PROVIDER_H
+
+class exprt;
+
+class conflict_providert
+{
+public:
+  /// Returns true if an expression is in the final conflict leading to UNSAT.
+  /// The argument can be a Boolean expression or something more
+  /// solver-specific, e.g. a `literal_exprt`.
+  virtual bool is_in_conflict(const exprt &) const = 0;
```

Comment:
Is it really worth making it this general when really we just have support for this in `prop_conv_solvert`? I'd vote for taking a `literal_exprt` here instead.

Reply:
I'd agree with that, but then I could as well leave it as `literalt`. Since we go for the more general `exprt` interfaces now, I think there is no harm keeping it more general.